src/Microsoft.Diagnostics.Monitoring.RestServer/Converters/IntegerOrHexJsonConverter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using Newtonsoft.Json;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer.Converters
+{
+    public class IntegerOrHexJsonConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(long) == objectType;
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (typeof(long) == reader.ValueType)
+            {
+                return (long)reader.Value;
+            }
+
+            string stringValue = reader.Value as string;
+            if (string.IsNullOrWhiteSpace(stringValue))
+            {
+                throw new JsonSerializationException("Value cannot be null, empty, or whitespace.");
+            }
+            else if (stringValue.StartsWith("0x", StringComparison.OrdinalIgnoreCase))
+            {
+                if (!long.TryParse(stringValue.Substring(2), NumberStyles.AllowHexSpecifier, CultureInfo.InvariantCulture, out long result))
```

Comment:
Your allowing the hex specifier and doing a substring to skip it.

Reply:
Yes, that is intetional. From https://docs.microsoft.com/en-us/dotnet/api/system.globalization.numberstyles?view=netcore-3.1: "Strings that are parsed using this style cannot be prefixed with "0x" or "&h"."